### PR TITLE
[pentest] Configure additional alerts for FI

### PR DESCRIPTION
**Pull request by nasahlpa**
This commit adds additional alerts to the alert handler configuration and fixes a bug in the configuration of the alert handler. Previously random alerts were triggered, now, only the alerts are reported that were triggered by a FI attack.

### CODE REVIEW

File: sw/device/tests/crypto/cryptotest/firmware/sca_lib.h [hunk 1]
```diff
@@ -5,8 +5,36 @@
 #ifndef OPENTITAN_SW_DEVICE_TESTS_CRYPTO_CRYPTOTEST_FIRMWARE_SCA_LIB_H_
 #define OPENTITAN_SW_DEVICE_TESTS_CRYPTO_CRYPTOTEST_FIRMWARE_SCA_LIB_H_
```
Comment (johannheyszl) [inline sw/device/tests/crypto/cryptotest/firmware/sca_lib.h:7]: I needed to add
#include <stdint.h>

**Review by johannheyszl — APPROVED** [review]
Thanks @nasahlpa - builds with the mini fix as comment

File: sw/device/tests/crypto/cryptotest/firmware/sca_lib.h (hunk 1 above)
Comment (nasahlpa) [reply sw/device/tests/crypto/cryptotest/firmware/sca_lib.h:7]: Ah thanks, done.
`#include "sw/device/lib/ujson/ujson.h"`
was missing.

**Review by vogelpi — APPROVED** [review]
LGTM

**Comment by jwnrt** [comment]
We're seeing CI failures on the ES branch that might be related to this PR

https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=148855&view=logs&jobId=5df5adf8-3577-5927-f0b2-31afed2697ee&j=5df5adf8-3577-5927-f0b2-31afed2697ee&t=dcb8db2a-d944-5acc-e50e-bea93f2a14e9

```
(13:27:25) ERROR: /azp/agent/_work/1/s/sw/device/tests/penetrationtests/firmware/BUILD:54:11: Compiling sw/device/tests/penetrationtests/firmware/ibex_fi.c failed: (Exit 1): clang failed: error executing command (from target //sw/device/tests/penetrationtests/firmware:ibex_fi) 
  (cd /root/.cache/bazel/_bazel_root/5e4f8c833cf3a422fdf6ca333dd151d7/sandbox/processwrapper-sandbox/8574/execroot/lowrisc_opentitan && \
  exec env - \
    BAZEL_CXXOPTS='-std=gnu++14' \
    PATH=/root/.cache/bazelisk/downloads/bazelbuild/bazel-6.2.1-linux-x86_64/bin:/tools/verilator/v4.210/bin:/tools/verible/bin:/root/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
  external/crt/toolchains/lowrisc_rv32imcb/wrappers/clang -MD -MF bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/tests/penetrationtests/firmware/_objs/ibex_fi/1/ibex_fi.d '-frandom-seed=bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/tests/penetrationtests/firmware/_objs/ibex_fi/1/ibex_fi.o' '-DBAZEL_CURRENT_REPOSITORY=""' -iquote . -iquote bazel-out/k8-fastbuild-ST-2cc462681f62/bin -iquote external/freertos -iquote bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/freertos -iquote external/manufacturer_test_hooks -iquote bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/manufacturer_test_hooks -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/flash_ctrl/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/rv_core_ibex/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/spi_device/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/uart/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/flash_ctrl/data/autogen -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/clkmgr/data/autogen -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/csrng/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/edn/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/entropy_src/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/gpio/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/pinmux/data/autogen -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/rv_timer/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/lc_ctrl/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/rstmgr/data/autogen -isystem sw/device/lib/testing/test_framework -isystem bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/lib/testing/test_framework -isystem external/freertos/include -isystem bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/freertos/include -isystem external/freertos/portable/GCC/RISC-V -isystem bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/freertos/portable/GCC/RISC-V -nostdinc -isystemexternal/lowrisc_rv32imcb_files/lib/clang/16/include -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include/c++/10.2.0 -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include/c++/10.2.0/backward -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include/c++/10.2.0/riscv32-unknown-elf '-march=rv32imc_zba_zbb_zbc_zbs' '-mabi=ilp32' '-mcmodel=medany' -mlittle-endian -Wall -Werror -Wextra -Wno-covered-switch-default -Wconversion -Wimplicit-fallthrough -Winvalid-pch -Wstrict-prototypes -Wswitch-default -Wno-missing-field-initializers -Wno-sign-compare '-Wno-error=unused-function' -Wno-unused-parameter '-Werror=date-time' -ffunction-sections -fdata-sections -Os -g -gdwarf-4 -ffreestanding -fno-common '' -Wgnu -Wno-gnu-zero-variadic-macro-arguments -Wno-gnu-statement-expression-from-macro-expansion -mguards '-std=gnu11' -c sw/device/tests/penetrationtests/firmware/ibex_fi.c -o bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/tests/penetrationtests/firmware/_objs/ibex_fi/1/ibex_fi.o)
# Configuration: 3521f93652866bf4b4eb6989f27a92c74b48a8c388637ccfb1e9c11d8f5732b6
# Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
sw/device/tests/penetrationtests/firmware/ibex_fi.c:489:12: error: initializing 'uint32_t' (aka 'unsigned int') with an expression of incompatible type 'sca_registered_alerts_t' (aka 'struct sca_registered_alerts')
  uint32_t reg_alerts = sca_get_triggered_alerts();
           ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~
sw/device/tests/penetrationtests/firmware/ibex_fi.c:507:14: error: assigning to 'uint32_t' (aka 'unsigned int') from incompatible type 'sca_registered_alerts_t' (aka 'struct sca_registered_alerts')
  reg_alerts = sca_get_triggered_alerts();
             ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~
2 errors generated.
```

Since this PR passed CI it might be a merge skew error.

**Comment by nasahlpa** [comment]
> We're seeing CI failures on the ES branch that might be related to this PR
> 
> https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=148855&view=logs&jobId=5df5adf8-3577-5927-f0b2-31afed2697ee&j=5df5adf8-3577-5927-f0b2-31afed2697ee&t=dcb8db2a-d944-5acc-e50e-bea93f2a14e9
> 
> ```
> (13:27:25) ERROR: /azp/agent/_work/1/s/sw/device/tests/penetrationtests/firmware/BUILD:54:11: Compiling sw/device/tests/penetrationtests/firmware/ibex_fi.c failed: (Exit 1): clang failed: error executing command (from target //sw/device/tests/penetrationtests/firmware:ibex_fi) 
>   (cd /root/.cache/bazel/_bazel_root/5e4f8c833cf3a422fdf6ca333dd151d7/sandbox/processwrapper-sandbox/8574/execroot/lowrisc_opentitan && \
>   exec env - \
>     BAZEL_CXXOPTS='-std=gnu++14' \
>     PATH=/root/.cache/bazelisk/downloads/bazelbuild/bazel-6.2.1-linux-x86_64/bin:/tools/verilator/v4.210/bin:/tools/verible/bin:/root/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
>     PWD=/proc/self/cwd \
>   external/crt/toolchains/lowrisc_rv32imcb/wrappers/clang -MD -MF bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/tests/penetrationtests/firmware/_objs/ibex_fi/1/ibex_fi.d '-frandom-seed=bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/tests/penetrationtests/firmware/_objs/ibex_fi/1/ibex_fi.o' '-DBAZEL_CURRENT_REPOSITORY=""' -iquote . -iquote bazel-out/k8-fastbuild-ST-2cc462681f62/bin -iquote external/freertos -iquote bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/freertos -iquote external/manufacturer_test_hooks -iquote bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/manufacturer_test_hooks -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/flash_ctrl/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/rv_core_ibex/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/spi_device/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/uart/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/flash_ctrl/data/autogen -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/clkmgr/data/autogen -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/csrng/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/edn/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/entropy_src/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/gpio/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/pinmux/data/autogen -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/rv_timer/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/ip/lc_ctrl/data -Ibazel-out/k8-fastbuild-ST-2cc462681f62/bin/hw/top_earlgrey/ip/rstmgr/data/autogen -isystem sw/device/lib/testing/test_framework -isystem bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/lib/testing/test_framework -isystem external/freertos/include -isystem bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/freertos/include -isystem external/freertos/portable/GCC/RISC-V -isystem bazel-out/k8-fastbuild-ST-2cc462681f62/bin/external/freertos/portable/GCC/RISC-V -nostdinc -isystemexternal/lowrisc_rv32imcb_files/lib/clang/16/include -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include/c++/10.2.0 -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include/c++/10.2.0/backward -isystemexternal/lowrisc_rv32imcb_files/riscv32-unknown-elf/include/c++/10.2.0/riscv32-unknown-elf '-march=rv32imc_zba_zbb_zbc_zbs' '-mabi=ilp32' '-mcmodel=medany' -mlittle-endian -Wall -Werror -Wextra -Wno-covered-switch-default -Wconversion -Wimplicit-fallthrough -Winvalid-pch -Wstrict-prototypes -Wswitch-default -Wno-missing-field-initializers -Wno-sign-compare '-Wno-error=unused-function' -Wno-unused-parameter '-Werror=date-time' -ffunction-sections -fdata-sections -Os -g -gdwarf-4 -ffreestanding -fno-common '' -Wgnu -Wno-gnu-zero-variadic-macro-arguments -Wno-gnu-statement-expression-from-macro-expansion -mguards '-std=gnu11' -c sw/device/tests/penetrationtests/firmware/ibex_fi.c -o bazel-out/k8-fastbuild-ST-2cc462681f62/bin/sw/device/tests/penetrationtests/firmware/_objs/ibex_fi/1/ibex_fi.o)
> # Configuration: 3521f93652866bf4b4eb6989f27a92c74b48a8c388637ccfb1e9c11d8f5732b6
> # Execution platform: @local_config_platform//:host
> 
> Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
> sw/device/tests/penetrationtests/firmware/ibex_fi.c:489:12: error: initializing 'uint32_t' (aka 'unsigned int') with an expression of incompatible type 'sca_registered_alerts_t' (aka 'struct sca_registered_alerts')
>   uint32_t reg_alerts = sca_get_triggered_alerts();
>            ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~
> sw/device/tests/penetrationtests/firmware/ibex_fi.c:507:14: error: assigning to 'uint32_t' (aka 'unsigned int') from incompatible type 'sca_registered_alerts_t' (aka 'struct sca_registered_alerts')
>   reg_alerts = sca_get_triggered_alerts();
>              ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~
> 2 errors generated.
> ```
> 
> Since this PR passed CI it might be a merge skew error.

This should be fixed with [this](https://github.com/lowRISC/opentitan/commit/7af759be6c99076d47e47fb96c333e854fd63dc3) commit.

**Pull request merged**